client/ayon_nuke/api/plugin.py

```diff
@@ -309,6 +311,30 @@ def set_selected_nodes(self, pre_create_data):
         else:
             self.selected_node = None
 
+    def update_instances(self, update_list):
+        for created_inst, changes in update_list:
+            instance_node = created_inst.transient_data["node"]
+
+            # update instance node name if product name changed
+            if "productName" in changes.changed_keys:
+                instance_node["name"].setValue(
+                    changes["productName"].new_value
+                )
+            # in case node is not existing anymore (user erased it manually)
+            try:
+                instance_node.fullName()
+            except ValueError:
+                self.remove_instances([created_inst])
+                continue
```

Comment:
This would also need to be moved after the node removal check because if it wouldn't exist I'm pretty sure this would error.
```suggestion
            # in case node is not existing anymore (user erased it manually)
            try:
                instance_node.fullName()
            except ValueError:
                self.remove_instances([created_inst])
                continue
            # update instance node name if product name changed
            if "productName" in changes.changed_keys:
                instance_node["name"].setValue(
                    changes["productName"].new_value
                )
```